cognite/client/_api/assets.py

```diff
@@ -268,6 +268,70 @@ def list(
             partitions=partitions,
         )
 
+    def aggregate(
+        self,
+        name: str = None,
+        parent_ids: List[int] = None,
+        parent_external_ids: List[str] = None,
+        asset_subtree_ids: List[int] = None,
+        asset_subtree_external_ids: List[str] = None,
+        data_set_ids: List[int] = None,
+        data_set_external_ids: List[str] = None,
+        metadata: Dict[str, str] = None,
+        source: str = None,
+        created_time: Union[Dict[str, Any], TimestampRange] = None,
+        last_updated_time: Union[Dict[str, Any], TimestampRange] = None,
+        root: bool = None,
```

Comment:
I was trying to look into future here, and as far as I know there will be other props in this query, like `groupBy` for instance. Because of this and because of the fact that it's hard to maintain this I was thinking to take filter as an object parameter (as it is in api).

@erlendvollset I am not sure if it's convenient enough for pydevs to pass nested params into function. could you apply a little bit of your judgement here?

Reply:
I've found a few places in code where it is done with the same approach, so I didn't think too much about it. For instance, list methods for timeseries, sequences, events, etc. are done in the same way.

Reply:
yep, you're right. list is basically just a filter endpoint, so it doesn't do anything else. but I am not sure if we should do the same here

Reply:
@sanderland do you have opinion on this one?

Reply:
you mean make it more like search rather than list? Does the API use the same filter spec as search? If so I think as long as you allow a dict for the parameter (again, like search) it's fine. it's a bit of a developer vs user convenience tradeoff. 
If not, you end up implementing another possibly autogenerated filter object which is incompatible with the other and will be confusing I think